### PR TITLE
[KOGITO-6282] Removed dependency. Fixed tests

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @evacchi  @jiripetrlik

See https://issues.redhat.com/browse/KOGITO-6282

Needed by

https://github.com/kiegroup/kogito-runtimes/pull/1735

### CODE REVIEW

**Review by jiripetrlik — APPROVED** [review]

File: dmn-drools-springboot-metrics/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # Configuration file
-# key = value
\ No newline at end of file
+# key = value
+management.security.enabled=false
+management.endpoints.web.exposure.include=metrics,prometheus
```
Comment (danielezonca) [inline dmn-drools-springboot-metrics/src/main/resources/application.properties]: I don't like that user has to configure this explicitly but if this is the suggested SpringBoot approach it is fine for me.

Can you please update Kogito documentation too?
If you search `/metrics` in the doc page https://docs.kogito.kie.org/latest/html_single/ there are multiple occurrences to update (the source of the doc is in this branch https://github.com/kiegroup/kie-docs/tree/main-kogito ). Please also mention to add these properties in the SpringBoot part.
I think it is useful to add a note in `1.15` [release notes](https://github.com/kiegroup/kie-docs/tree/main-kogito/doc-content/kogito-docs/src/main/asciidoc/release-notes) that now the metrics are exposed in a different endpoint.

**Comment by danielezonca** [comment]
@radtriste 
This PRs changes the exposed metrics endpoint. Do you know if we have any image/BDD test that needs to be updated too? I don't remember if we can trigger downstream BDD tests from PR 🤔

File: dmn-drools-quarkus-metrics/docker-compose.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ services:
       - hello
   grafana:
     image: grafana/grafana:6.6.1
+    user: "$UID:$GID"
```
Comment (danielezonca) [inline dmn-drools-quarkus-metrics/docker-compose.yml]: Can you please explain this change?
If I run it on mac I got
```
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 
```

**Comment by radtriste** [comment]
> @radtriste This PRs changes the exposed metrics endpoint. Do you know if we have any image/BDD test that needs to be updated too? I don't remember if we can trigger downstream BDD tests from PR thinking

@danielezonca we have BDD test integration with Prometheus. I guess we need to update the operator.
paths would be `/q/metrics` and `/actuator/metrics` by default right ?
cc @spolti @vaibhavjainwiz for confirmation

**Comment by gitgabrio** [comment]
@radtriste 
/actuator/prometheus

**Comment by vaibhavjainwiz** [comment]
@radtriste yes, metric endpoint in kogito-operator needs to be updated with this change. I had created below jira for it
https://issues.redhat.com/browse/KOGITO-6293

**Comment by gitgabrio** [comment]
@danielezonca @vaibhavjainwiz 
See https://github.com/kiegroup/kogito-runtimes/pull/1735#issuecomment-976482492

**Comment by danielezonca** [comment]
> @danielezonca @vaibhavjainwiz See [kiegroup/kogito-runtimes#1735 (comment)](https://github.com/kiegroup/kogito-runtimes/pull/1735#issuecomment-976482492)

This change has been reverted so this comment is not valid anymore

> @radtriste yes, metric endpoint in kogito-operator needs to be updated with this change. I had created below jira for it https://issues.redhat.com/browse/KOGITO-6293

@vaibhavjainwiz 
How do you want to proceed? Should we wait the update of the operator to merge everything together or can we proceed and merge?
We can keep the old endpoint as backward compatible change if you plan to update the operator in the future and not now so up to you :)

**Comment by vaibhavjainwiz** [comment]
> > @danielezonca @vaibhavjainwiz See [kiegroup/kogito-runtimes#1735 (comment)](https://github.com/kiegroup/kogito-runtimes/pull/1735#issuecomment-976482492)
> 
> This change has been reverted so this comment is not valid anymore
> 
> > @radtriste yes, metric endpoint in kogito-operator needs to be updated with this change. I had created below jira for it https://issues.redhat.com/browse/KOGITO-6293
> 
> @vaibhavjainwiz How do you want to proceed? Should we wait the update of the operator to merge everything together or can we proceed and merge? We can keep the old endpoint as backward compatible change if you plan to update the operator in the future and not now so up to you :)

@danielezonca Kogito-operator PR is ready for these changes(https://github.com/kiegroup/kogito-operator/pull/1081). We could merge operator changes as soon as kogito-runtime PR(https://github.com/kiegroup/kogito-runtimes/pull/1735) will merge.
Now we could merge everything togeather.
Cc. @amygaoo

**Comment by radtriste** [comment]
> > > @danielezonca @vaibhavjainwiz See [kiegroup/kogito-runtimes#1735 (comment)](https://github.com/kiegroup/kogito-runtimes/pull/1735#issuecomment-976482492)
> > 
> > 
> > This change has been reverted so this comment is not valid anymore
> > > @radtriste yes, metric endpoint in kogito-operator needs to be updated with this change. I had created below jira for it https://issues.redhat.com/browse/KOGITO-6293
> > 
> > 
> > @vaibhavjainwiz How do you want to proceed? Should we wait the update of the operator to merge everything together or can we proceed and merge? We can keep the old endpoint as backward compatible change if you plan to update the operator in the future and not now so up to you :)
> 
> @danielezonca Kogito-operator PR is ready for these changes([kiegroup/kogito-operator#1081](https://github.com/kiegroup/kogito-operator/pull/1081)). We could merge operator changes as soon as kogito-runtime PR([kiegroup/kogito-runtimes#1735](https://github.com/kiegroup/kogito-runtimes/pull/1735)) will merge. Now we could merge everything togeather. Cc. @amygaoo

Please leave 24 hours between the 2 merges (runtimes/examples and then operator) so that artifacts can be deployed during the nightly in-between ;)

**Comment by radtriste** [comment]
what is the status there ? do we want that in 1.15 ?

**Review by danielezonca — APPROVED** [review]

**Comment by danielezonca** [comment]
> what is the status there ? do we want that in 1.15 ?

Yes it is ready to be merged 👍

**Comment by danielezonca** [comment]
@gitgabrio 
Can you fix the conflict?

**Comment by gitgabrio** [comment]
@danielezonca @vaibhavjainwiz: The dependent jira (https://issues.redhat.com/browse/KOGITO-6293) does not need this one. The new metrics endpoints have been created in a previous Ticket (https://issues.redhat.com/browse/KOGITO-5931), so switch from "/metrics"  to "/q/metrics" or "/actuator/prometheus" should already work in master. This PR is only about removing the "/metrics" endpoint, it does not add anything new

**Comment by gitgabrio** [comment]
> @gitgabrio Can you fix the conflict?

 @danielezonca done

**Pull request merged**